src/main/java/com/bpodgursky/jbool_expressions/NExpression.java

```diff
@@ -12,14 +12,19 @@
 
   public final Expression<K>[] expressions;
 
-  protected NExpression(List<? extends Expression<K>> expressions){
+  protected NExpression(List<? extends Expression<K>> expressions, long hashBase){
```

Comment:
I was kind of thinking hashBase could be an object and NEXpression could call .hashCode, saving a line or two

Reply:
wouldn't `theObject.hashCode()` return `int` though?